pydra/engine/tests/test_dockertask.py

```diff
@@ -757,15 +758,24 @@ def test_docker_inputspec_2(plugin, tmpdir):
             (
                 "file1",
                 attr.ib(
-                    type=File, metadata={"position": 1, "help_string": "input file 1"}
+                    type=File,
+                    metadata={
+                        "position": 1,
+                        "argstr": "",
```

Comment:
what does this do? is this equivalent to `"%s"` in nipype 1.   

you can imagine that position and/or argstr as necessary. 
- if position is given without argstr, it can default to `argstr = "{variablename}"`
- if no position and no argstr it doesn't go on the command line
- if no position then argstr must be `argstr = "{variablename}"` to be used. i think this makes it explicit.

Reply:
`argstr` can be `-f`, `-f...`, `-f {variable}`, or `-f {variable}...`, if nothing is provided, than only `variable` is used, so yes you can treat  `argstr = "{variablename}"`

I don't treat `position` and `argstr` together. There are completely independent for me. Position can be provided or not. If not provided than I assigned the position.